kong/plugins/key-auth/handler.lua

```diff
@@ -25,6 +25,11 @@ local function load_credential(key)
     return nil, err
   end
 
+  if cred.ttl == 0 then
+    kong.log.debug("key expired")
+    return nil, err
```

Comment:
In this case `err` is nil?

Reply:
This `err` might be `nil` or not depending on the `mlcache` backend.